dbms/CMakeLists.txt

```diff
@@ -564,11 +564,14 @@ if (USE_JEMALLOC)
     dbms_target_include_directories (SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # used in Interpreters/AsynchronousMetrics.cpp
     target_include_directories (clickhouse_new_delete SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR})
 
-    if(NOT MAKE_STATIC_LIBRARIES AND ${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$")
-        # mallctl in dbms/src/Interpreters/AsynchronousMetrics.cpp
-        # Actually we link JEMALLOC to almost all libraries.
-        # This is just hotfix for some uninvestigated problem.
-        target_link_libraries(clickhouse_interpreters PRIVATE ${JEMALLOC_LIBRARIES})
+    if(NOT MAKE_STATIC_LIBRARIES)
+        target_link_libraries(clickhouse_common_io PRIVATE ${JEMALLOC_LIBRARIES})
+	    if (${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$")
+            # mallctl in dbms/src/Interpreters/AsynchronousMetrics.cpp
+            # Actually we link JEMALLOC to almost all libraries.
+            # This is just hotfix for some uninvestigated problem.
+            target_link_libraries(clickhouse_interpreters PRIVATE ${JEMALLOC_LIBRARIES})
+        endif()
```

Comment:
Actually looks like this is unnecessary already (either after `-Wl,--unresolved-symbols=ignore-all` or smth else, not sure).
And indeed, I tried various unbunlded environments (splitted, not splitted, static/shared/internal jemalloc) and everything works (refs 6cb5d0046beb37c2146309fda5b022579dd4893e).

As for the issue, seems that the better alternative will be:

```patch
 if (MAKE_STATIC_LIBRARIES OR NOT SPLIT_SHARED_LIBRARIES)
     add_library (dbms STATIC ${dbms_headers} ${dbms_sources})
     set (all_modules dbms)
+    target_link_libraries (dbms PRIVATE -Wl,--unresolved-symbols=ignore-all)
 else()
```

@amosbird ? (you are the one who added it for splitted build in 9152a832c47)